cpp/include/DataStorm/InternalT.h

```diff
@@ -439,13 +439,18 @@ namespace DataStormI
 
         template<typename FF> void init(const std::string& name, FF&& lambda)
         {
-            _name = name;
-            _lambda = std::forward<FF>(lambda);
+            std::unique_lock lock(_mutex);
+            if (!_lambda)
```

Comment:
The logic is init can be called multiple times? This deserves a comment since it's unexpected with a function named `init`.